client/state/sites/plans/selectors.js

```diff
@@ -46,10 +47,44 @@ export const getCurrentPlan = ( state, siteId ) => {
 		debug( 'current plan: %o', plan );
 		return plan;
 	}
-
 	return null;
 };
 
+/**
+ * Returns a site specific plan
+ * @param  {Object} state        global state
+ * @param  {Number} siteId       the site id
+ * @param  {String} productSlug  the plan product slug
+ * @return {Object} the matching plan
+ */
+export const getSitePlan = createSelector(
+	( state, siteId, productSlug ) => {
+		const plansBySiteId = getPlansBySiteId( state, siteId );
+		if ( ! plansBySiteId || plansBySiteId.isRequesting || ! plansBySiteId.data ) {
+			return null;
+		}
+		return plansBySiteId.data.filter( plan => plan.productSlug === productSlug ).shift();
+	},
+	( state, siteId ) => getPlansBySiteId( state, siteId )
+);
+
+/**
+ * Returns a plan discount price
+ * @param  {Object}  state         global state
+ * @param  {Number}  siteId       the site id
+ * @param  {String}  productSlug   the plan product slug
+ * @param  {Boolean} isMonthly     if true, returns monthly price
+ * @return {Number}  plan discount price
+ */
+export function getPlanDiscountPrice( state, siteId, productSlug, isMonthly = false ) {
+	const plan = getSitePlan( state, siteId, productSlug );
+	if ( get( plan, 'rawPrice', -1 ) < 0 || get( plan, 'rawDiscount', -1 ) <= 0 ) {
+		return null;
+	}
+	const discountPrice = plan.rawPrice;
+	return isMonthly ? parseFloat( ( discountPrice / 12 ).toFixed( 2 ) ) : discountPrice;
```

Comment:
Just nitpicking, but I noticed (also in other PRs) you don't do much newlines/code separation. E.g.:

```js
export function getPlanDiscountPrice( state, siteId, productSlug, isMonthly = false ) {
	const plan = getSitePlan( state, siteId, productSlug );

	if ( get( plan, 'rawPrice', -1 ) < 0 || get( plan, 'rawDiscount', -1 ) <= 0 ) {
		return null;
	}

	const discountPrice = plan.rawPrice;

	return isMonthly ? parseFloat( ( discountPrice / 12 ).toFixed( 2 ) ) : discountPrice;
}
```

is much more readable for me and the code can breathe. I know this is a detail and everyone has her/his own style, but I think at least variable definitions could be on separate line and then one new line after the `if` condition and one new line before `return` statement.

What do you think? :)